### ISSUE TITLE
2019_09_12 Cloned Source Fails to Build on openSUSE Linux

### ISSUE THREAD

**Issue by martinvahi** [JulNadeauCA/libagar#34]
The whole console output MIGHT be available from 

https://temporary.softf1.com/2019/bugs/2019_09_12_Agar_Fails_to_Compile_bugreport_t1.txt

but the first error seems to be

    gcc -M  -march=native -D_AGAR_INTERNAL -D_DEFAULT_SOURCE -D_BSD_SOURCE -I/home/ts2/m_local/lib/Agar/kompil/libagar/include -D_XOPEN_SOURCE=600 -D_AGAR_CORE_INTERNAL -Ix86_64-unknown-linux-gnu byteswap.c class.c config.c core.c cpuinfo.c data_source.c db.c dir.c dso.c error.c event.c exec.c file.c getopt.c load_integral.c load_real.c load_string.c load_version.c object.c string.c tbl.c text.c time.c time_dummy.c timeout.c threads.c vasprintf.c vsnprintf.c user.c user_dummy.c user_getenv.c variable.c vec.c time_posix.c time_renderer.c time_gettimeofday.c net.c net_dummy.c net_bsd.c user_posix.c
    In file included from /usr/include/sys/types.h:132:0,
                     from /usr/include/stdlib.h:314,
                     from /home/ts2/m_local/lib/Agar/kompil/libagar/include/agar/core/core.h:40,
                     from byteswap.c:7:
    ./time.h:4:3: error: #error "Must be included by object.h"
     # error "Must be included by object.h"
       ^~~~~
    In file included from /usr/include/sys/select.h:43:0,

I do admit that I  am running relatively old openSUSE version, "openSUSE Leap"  version 42.3, but that should not be a problem for a GUI library that aims to be portable.

**Comment by martinvahi** [comment]
No luck. The new console output MIGHT be available from 

https://temporary.softf1.com/2019/bugs/2019_09_14_Agar_Fails_to_Compile_bugreport_t1_compilation_attempt_01.txt

The first error line seems to be

    gcc -march=native -ftree-vectorize  -D_AGAR_INTERNAL -D_DEFAULT_SOURCE -D_BSD_SOURCE  -I/home/ts2/m_local/lib/Agar/kompil/libagar/include -U_XOPEN_SOURCE -D_XOPEN_SOURCE=600        -I/usr/include -D_AGAR_CORE_INTERNAL  -fPIC -o string.o -c string.c
    string.c:76:35: error: 'AG_STRING_BUFFERS_MAX' undeclared here (not in a function); did you mean 'AG_BUFFER_MAX'?
     static char *_Nullable agPrintBuf[AG_STRING_BUFFERS_MAX];
                                       ^~~~~~~~~~~~~~~~~~~~~
                                       AG_BUFFER_MAX
    string.c:80:8: error: unknown type name 'AG_FmtStringExt'
     static AG_FmtStringExt *_Nullable agFmtExtensions = NULL;
            ^~~~~~~~~~~~~~~
    string.c:111:9: error: unknown type name 'AG_FmtString'; did you mean 'AG_VecString'?
     PrintU8(AG_FmtString *_Nonnull fs, char *_Nonnull dst, AG_Size dstSize)
             ^~~~~~~~~~~~
             AG_VecString

Would it make sense if I tried to set up a 

    https://travis-ci.org/

for the Agar? As of 2019_09_14 I do not know, if they have openSUSE Linux option but I suspect that for a start almost any distro would do with the libagar.

**Comment by JulNadeauCA** [comment]
Yes, that would be very helpful. I tried to get a recent OpenSUSE running under VirtualBox but it locked up during installation for some reason. I could not reproduce this in other distros.

AG_STRING_BUFFERS_MAX is in core/string.h:40, which is included by core/string.c. Those errors:

  class.c:65:11: warning: incompatible implicit declaration of built-in function 'strrchr'
  class.c:65:11: note: include '<string.h>' or provide a declaration of 'strrchr'

are very odd, because core/class.c does include <string.h> on line 33.

I wonder why is the compiler apparently confusing agar's core/string.h with the system string.h (as it did with <time.h> until I renamed agar's core/time.h to core/agtime.h)? The compiler seems to be behaving as if /home/ts2/m_local/lib/Agar/kompil/libagar/include/agar/core was added as a standard include (-I) path, which should never be the case.
